src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/Rules/AssemblyIdentityMustMatchTests.cs

```diff
@@ -0,0 +1,247 @@
+            string rightDllPath = Path.Combine(buildCommand.GetOutputDirectory("net6.0").FullName, "Project.dll");
+            IAssemblySymbol leftSymbols = new AssemblySymbolLoader().LoadAssembly(leftDllPath);
+            IAssemblySymbol rightSymbols = new AssemblySymbolLoader().LoadAssembly(rightDllPath);
+
+            ApiComparer differ = new();
+            IEnumerable<CompatDifference> differences = differ.GetDifferences(leftSymbols, rightSymbols);
+
+            // net6.0 assembly should have same or higher version than net6.0
+            Assert.Single(differences);
+            Assert.Equal("CP0003 : left assembly version '2.0.0.0' should be not higher than right version '0.0.0.0'.", differences.First().ToString());
+        }
+
+        [Fact]
+        public void AssemblyVersionMustBeStrictlyCompatible()
+        {
+            // setting different assembly culture for netstanard2.0
+            string assemblyCulture = "#if !NET6_0 \n[assembly: System.Reflection.AssemblyVersionAttribute(\"1.0.0.0\")]\n #else\n[assembly: System.Reflection.AssemblyVersionAttribute(\"2.0.0.0\")]\n #endif";
+
+            TestProject testProject = new TestProject()
+            {
+                Name = "Project",
+                IsExe = false,
+                TargetFrameworks = "net6.0;netstandard2.0",
+            };
+
+            testProject.SourceFiles.Add("AssemblyInfo.cs", assemblyCulture);
+            testProject.AdditionalProperties["GenerateAssemblyInfo"] = "false";
+
+            TestAsset testAsset = _testAssetsManager.CreateTestProject(testProject);
+            BuildCommand buildCommand = new(testAsset);
+            buildCommand.Execute().Should().Pass();
+            string leftDllPath = Path.Combine(buildCommand.GetOutputDirectory("netstandard2.0").FullName, "Project.dll");
+            string rightDllPath = Path.Combine(buildCommand.GetOutputDirectory("net6.0").FullName, "Project.dll");
+            IAssemblySymbol leftSymbols = new AssemblySymbolLoader().LoadAssembly(leftDllPath);
+            IAssemblySymbol rightSymbols = new AssemblySymbolLoader().LoadAssembly(rightDllPath);
+
+            // Compatible assembly versions
+            ApiComparer differ = new();
+            IEnumerable<CompatDifference> differences = differ.GetDifferences(leftSymbols, rightSymbols);
+            Assert.Empty(differences);
+
+            ApiComparer strictDiffer = new();
+            strictDiffer.StrictMode = true;
+            leftSymbols = new AssemblySymbolLoader().LoadAssembly(leftDllPath);
+            rightSymbols = new AssemblySymbolLoader().LoadAssembly(rightDllPath);
+
+            // Not strictly compatible
+            differences = strictDiffer.GetDifferences(leftSymbols, rightSymbols);
+            Assert.Single(differences);
+            Assert.Equal("CP0003 : right assembly version '2.0.0.0' should be not higher than left version '1.0.0.0'.", differences.First().ToString());
```

Comment:
instead of doing `differences.First().ToString()` since the other properties in CompatDifference matter, let's assert for those as well?